### PR TITLE
NxToggle gapless RSC-484 

### PR DESCRIPTION
**Pull request by glenhunter**
https://issues.sonatype.org/browse/RSC-484

In the end I went with "gap" because I thought that changing the `max-width` of the control would be a mistake if the label was long.

### CODE REVIEW

File: gallery/src/components/NxToggle/NxToggleGaplessExample.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useState } from 'react';
+
+import { NxToggle } from '@sonatype/react-shared-components';
+
+function NxToggleExample() {
+  // this example uses the `useState` hook for succinctness, but you could also manage the state manually
+  // in a class component
```
Comment (rpokorny) [inline gallery/src/components/NxToggle/NxToggleGaplessExample.tsx:13]: This comment is not something that needs to be duplicated in new copy/pasted examples.  In fact if anything I think the react community has moved far enough beyond class components that I'd rather not have this comment anywhere

File: gallery/src/components/NxToggle/NxToggleGaplessExample.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      </NxToggle>
+      <NxToggle className="nx-toggle--no-gap" inputId="no-label-check" onChange={toggleKraken} isChecked={isKrakenOut}>
+        Release all the Krakens immediately
+      </NxToggle>
+      <NxToggle className="nx-toggle--no-gap" inputId="children-check" onChange={toggleShapes} isChecked={isShapes}>
+        Allow shapes - like a circle, a perfectly round SVG circle, pleasing to the eye, not too big and not too
+        small, just right to appear beside a checkbox and demonstrate that the label wraps
+      </NxToggle>
+      <p>
+        {isWarpOn && 'Warp drive started'} {isKrakenOut && ' The Kraken is out!'}
```
Comment (rpokorny) [inline gallery/src/components/NxToggle/NxToggleGaplessExample.tsx:34]: Should this paragraph have a section based on `isShapes`?

File: gallery/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@sonatype/react-shared-components-gallery",
-  "version": "3.6.1",
+  "version": "3.6.2",
```
Comment (rpokorny) [inline gallery/package.json:3]: Should be a minor version

File: gallery/src/components/NxToggle/NxTogglePage.tsx [hunk 4]
```diff
@@ -84,6 +109,12 @@ const NxTogglePage = () =>
       This example shows a series of toggle controls in a typical vertical layout with
       different label content. Note that one of the toggle controls is disabled.
     </GalleryExampleTile>
+    <GalleryExampleTile title="General NxToggle Gapless Example"
+                        id="nx-toggle-gapless-example"
+                        codeExamples={exampleGaplessCode}
+                        liveExample={NxToggleGaplessExample}>
+      This example demonstrates a toggle control with only a small spacing gap between the label and the toggle control.
```
Comment (rpokorny) [inline gallery/src/components/NxToggle/NxTogglePage.tsx:116]:
```suggestion
      This example demonstrates toggle controls with only a small spacing gap between the label and the toggle control.
```

**Review by rpokorny — APPROVED** [review]
+1 aside from minor comments

File: gallery/package.json (hunk 3 above)
Comment (glenhunter) [reply gallery/package.json]: > Should be a minor version

I'm not always certain when it's a minor modification to an existing component: https://github.com/sonatype/sonatype-react-shared-components/pull/309/commits/3885453a58bba05042c563730145acc81412887b

File: gallery/src/components/NxToggle/NxToggleGaplessExample.tsx (hunk 2 above)
Comment (glenhunter) [reply gallery/src/components/NxToggle/NxToggleGaplessExample.tsx]: > Should this paragraph have a section based on `isShapes`?

Ah yes it should: https://github.com/sonatype/sonatype-react-shared-components/pull/309/commits/3885453a58bba05042c563730145acc81412887b

File: gallery/src/components/NxToggle/NxToggleGaplessExample.tsx (hunk 1 above)
Comment (glenhunter) [reply gallery/src/components/NxToggle/NxToggleGaplessExample.tsx]: > This comment is not something that needs to be duplicated in new copy/pasted examples. In fact if anything I think the react community has moved far enough beyond class components that I'd rather not have this comment anywhere

Removed here should we have a ticket for removing these comment more broadly? 

https://github.com/sonatype/sonatype-react-shared-components/pull/309/commits/3885453a58bba05042c563730145acc81412887b

**Pull request merged**